var/spack/repos/builtin/packages/hevea/package.py

```diff
@@ -28,6 +29,7 @@ class Hevea(MakefilePackage):
     # Dependency demands ocamlbuild
     depends_on("ocaml")
     depends_on("ocamlbuild")
+    depends_on("ocaml@4.0.0:4.99.9", when="@:2.35")
```

Comment:
This is different from the previous conflict, since it also imposes a lower bound (`ocaml` must be at major version 4). If this is the correct directive we can use just:
```suggestion
    depends_on("ocaml@4", when="@:2.35")
```
Also, shouldn't we impose a lower bound regardless of versions? It seems odd that `hevea@2.36` can use `ocaml@3` according to the directives.

Reply:
That's right. BTW prior to ocaml 4.8.0 configure should not set CC={self.compiler.cc}.
Maybe I shall open a PR for that later on.